cosmic_ray/util.py

```diff
@@ -64,5 +64,7 @@ def build_mutations(ops, to_ops):
     return [
         (idx, to_op)
         for idx, from_op in enumerate(ops)
-        for to_op in to_ops(from_op)
+        # note: mutations to self are excluded.
+        # None is a special mutation meaning to delete the operator!
+        for to_op in to_ops(from_op) if to_op is None or not isinstance(from_op, to_op)
```

Comment:
from_op is never None b/c this is the operator we are going to mutate. to_op is the new operator we want to use. In some cases (e.g. unary_operator_replacement.py) we also want to delete the operator, instead of replacing it with another value. A replacement of None signifies operator deletion.